### DISCUSSION TITLE
Display grain ID on IPDF

### DISCUSSION THREAD

**Discussion by bibiboom89** [mtex-toolbox/mtex#2288] [Ask Anything]
Hi everyone,

First of all, I’d like to wish you all a Happy New Year! 🎉

1. Short version: is there a way to find the coordinates xy in the ipf triangle of a given orientation ?
Right now, I'm converting my orientations into miller index which provides me xyz coordinates, but they are not exactly the same as in the IPF. I guess these is due to the projection distortion. But how can I correct that ?
```
  colorsG = ipfKey.orientation2color(ori(gSEL));
  rSym = symmetrise(vector3d.Y,ori.SS);
  h = ori(gSEL) \ rSym; h2 = project2FundamentalRegion(h);
  figure; h2.plot(colorsG,'fundamentalRegion','MarkerSize',MS,'FontSize',FS);
          text(h2.xyz(:,2),-h2.xyz(:,1),id)
```
![image](https://github.com/user-attachments/assets/34d6df13-7ecf-41f0-84ed-39e7b5675493)

2. Long version:
I’m currently working on a project where I’m trying to plot selected grains from an EBSD map onto an IPDF figure.
It can be found below two figures:
    The first image shows the selected grains from the EBSD map along with their identification numbers.
    The second image plots these grains on the IPDF triangle, visualized with the Young’s modulus isovalues.
My goal is to add the identification numbers to each dot on the IPDF figure so they can be easily referenced between the two figures. However, I couldn’t find a way to achieve this using MTEX.
Is there a way to retrieve dots positions from an IPDF figure so I can then display texts next to them ?

If anyone could guide me toward a solution or share relevant examples, I’d be incredibly grateful! 😊

Thank you in advance for your help!
Best regards,
Remy

![3s2_Stiched_with_matlab_Z1-Crystal-polished](https://github.com/user-attachments/assets/5d366bfd-e037-4d9c-83cb-ef14ec404a2c)
![3s2_Stiched_with_matlab_Z1-IPF_EY](https://github.com/user-attachments/assets/fb7d4eb0-56ce-4f4e-ae59-2b10aacf3bfa)

**Comment by kilir** [comment]
Hi,
how about the option 'label'?
```Matlab
o = orientation.rand(4,crystalSymmetry('622'))
labels = num2cell(num2str([1:length(o)]'))
plotIPDF(o,xvector,'label',labels)
```
where `labels` is a cell array of some characters.

Hope this helps.

Cheers,

Rüdiger

**Reply by bibiboom89** [reply]
Hi,

I was going to say: "oh this is so simple. How did I miss that.".
And then I saw that it only works if no color or values are used and this is why I thought it didn't work. 
So what I did is calling plotIPDF twice, once with values, second with labels and marker size put to 0.0001.
It works.
```
o = orientation.rand(4,crystalSymmetry('622'))
labels = num2cell(num2str([1:length(o)]'))
plotIPDF(o,1:4,xvector); hold on
plotIPDF(o,xvector,'label',labels,'MarkerSize',0.001)
```
![image](https://github.com/user-attachments/assets/63a2cb1f-0f5f-4162-b745-c2b6b2a92a8e)


Is there a way to control their positions, though?
The labels are slightly under the dots and gets on top with bigger font size.
I would like them to be just above the dots.

Thank you in advance for your help!
Best regards,
Remy
